includes/Pages/PageEditComment.php

```diff
@@ -121,4 +110,76 @@ protected function main()
             $this->setTemplate('edit-comment.tpl');
         }
     }
+
+    /**
+     * @param $comment
+     * @param $currentUser
+     *
+     * @return void
+     * @throws AccessDeniedException
+     */
+    protected function checkCommentAccess($comment, $currentUser): void
+    {
+        if ($comment->getUser() !== $currentUser->getId() && !$this->barrierTest('editOthers', $currentUser)) {
+            throw new AccessDeniedException($this->getSecurityManager(), $this->getDomainAccessManager());
+        }
+
+        $restrictedVisibility = $comment->getFlagged()
+            || $comment->getVisibility() === 'admin'
+            || $comment->getVisibility() === 'checkuser';
+
+        if ($restrictedVisibility && !$this->barrierTest('alwaysSeePrivateData', $currentUser, 'RequestData')) {
+            // Restricted visibility comments can only be seen if the user has a request reserved.
+            /** @var Request $request */
+            $request = Request::getById($comment->getRequest(), $comment->getDatabase());
+
+            if ($request->getReserved() !== $currentUser->getId()) {
+                throw new AccessDeniedException($this->getSecurityManager(), $this->getDomainAccessManager());
+            }
+        }
+
+        if ($comment->getVisibility() === 'admin'
+            && !$this->barrierTest('seeRestrictedComments', $currentUser, 'RequestData')
+            && $comment->getUser() !== $currentUser->getId()) {
+            throw new AccessDeniedException($this->getSecurityManager(), $this->getDomainAccessManager());
+        }
+
+        if ($comment->getVisibility() === 'checkuser'
+            && !$this->barrierTest('seeCheckuserComments', $currentUser, 'RequestData')
+            && $comment->getUser() !== $currentUser->getId()) {
+            throw new AccessDeniedException($this->getSecurityManager(), $this->getDomainAccessManager());
+        }
+    }
+
+    /**
+     * @param             $comment
+     * @param string|null $visibility
+     * @param             $newComment
+     *
+     * @throws ApplicationLogicException
+     */
+    protected function updateCommentData(Comment $comment, ?string $visibility, string $newComment): void
```

Comment:
The security of this function should be checked from the inside and not allow implementation to possibly breach security.

Reply:
I'm really not understanding what you're suggesting I change here.

This function should indeed be `private` rather than `protected` - I'll fix that later - but your comment seems to suggest something more substantial?